mm2src/coins/my_tx_history_v2.rs

```diff
@@ -406,22 +415,62 @@ where
         .get_history(&wallet_id, filters, request.paging_options.clone(), request.limit)
         .await?;
 
-    let transactions = history
-        .transactions
-        .into_iter()
-        .map(|mut details| {
-            // it can be the platform ticker instead of the token ticker for a pre-saved record
-            if details.coin != request.coin {
-                details.coin = request.coin.clone();
-            }
-            let confirmations = if details.block_height == 0 || details.block_height > current_block {
-                0
-            } else {
-                current_block + 1 - details.block_height
-            };
-            MyTxHistoryDetails { confirmations, details }
-        })
-        .collect();
+    let coin_conf = coin_conf(&ctx, coin.ticker());
+    let protocol_type = coin_conf["protocol"]["type"].as_str().unwrap_or_default();
```

Comment:
Instead of getting the protocol type from config, why not add `fn requires_remapping(&self) -> bool` to `CoinWithTxHistoryV2` and return true for Tendermint coin/asset.

Reply:
There also could be `CoinWithTxHistoryV2::remap_tx_if_required(&self, details: &mut TransactionDetails)`. So this remapping logic is performed on the coin's level.
What do you think?

Reply:
Agree, this is better :)

Reply:
> There also could be `CoinWithTxHistoryV2::remap_tx_if_required(&self, details: &mut TransactionDetails)`. So this remapping logic is performed on the coin's level. What do you think?

This is actually pretty good idea. However, we already have mapping for all v2 history implementations. So I will need to do this change for all. Should I cover it in this PR? I left a `TODO` note here
https://github.com/KomodoPlatform/atomicDEX-API/blob/452fc50b38ef4b73f6d20c058fb1b637a7885d0d/mm2src/coins/my_tx_history_v2.rs#L439-L443

Please let me know if this change is blocker and I should cover it.

Reply:
This is a non-blocker, you can do it in the next PR for sure.